src/App.tsx

```diff
@@ -1,17 +1,63 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { getVisibleTodos } from './helper';
+import { Filter } from './types/Filter';
+import { NewTodo } from './components/NewTodo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.all);
+  const [isError, setIsError] = useState(false);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleFilter = useCallback((str: Filter) => setFilter(str), []);
+  const handleError = () => {
```

Comment:
Also will be cool if this function get error message

```suggestion
  const showErrorMessage = () => {
```